File: frontend/src/api/testReportTemplate.ts
import request from './request'
import type { PageResult } from './testReport'

export interface TestReportTemplateRequestDTO {
    templateName: string
    templateType: 'EXECUTION' | 'COVERAGE' | 'QUALITY' | 'RISK'
    templateContent: string
    templateVariables?: string
    fileFormat?: 'WORD' | 'PDF' | 'EXCEL'
    templateDescription?: string
    isDefault?: '0' | '1'
    isActive?: '0' | '1'
    creatorId?: number
}

export interface TestReportTemplateResponseDTO {
    id: number
    templateCode: string
    templateName: string
    templateType: string
    templateContent: string
    templateVariables?: string
    fileFormat?: string
    templateDescription?: string
    isDefault: '0' | '1'
    isActive: '0' | '1'
    version?: number
    creatorId?: number
    createTime?: string
    updateTime?: string
}

export const testReportTemplateApi = {
    // Create template
    createTemplate(data: TestReportTemplateRequestDTO) {
        return request.post<any, TestReportTemplateResponseDTO>('/v1/test-report-templates', data)
    },

    // Get template list
    getTemplateList(page: number = 0, size: number = 10) {
        return request.get<any, PageResult<TestReportTemplateResponseDTO>>('/v1/test-report-templates', {
            params: { page, size }
        })
    },

    // Get template detail
    getTemplateById(id: number) {
        return request.get<any, TestReportTemplateResponseDTO>(`/v1/test-report-templates/${id}`)
    },

    // Get template by code
    getTemplateByCode(templateCode: string) {
        return request.get<any, TestReportTemplateResponseDTO>(`/v1/test-report-templates/code/${templateCode}`)
    },

    // Update template
    updateTemplate(id: number, data: TestReportTemplateRequestDTO) {
        return request.put<any, TestReportTemplateResponseDTO>(`/v1/test-report-templates/${id}`, data)
    },

    // Delete template
    deleteTemplate(id: number) {
        return request.delete<any, void>(`/v1/test-report-templates/${id}`)
    },

    // Toggle template status
    toggleTemplateStatus(id: number, isActive: '0' | '1') {
        return request.put<any, TestReportTemplateResponseDTO>(`/v1/test-report-templates/${id}/status`, null, {
            params: { isActive }
        })
    },

    // Set default template
    setDefaultTemplate(id: number, templateType: string) {
        return request.put<any, TestReportTemplateResponseDTO>(`/v1/test-report-templates/${id}/default`, null, {
            params: { templateType }
        })
    },

    // Get active templates by type
    getActiveTemplatesByType(templateType: string) {
        return request.get<any, TestReportTemplateResponseDTO[]>('/v1/test-report-templates/active', {
            params: { templateType }
        })
    },

    // Get default template by type
    getDefaultTemplateByType(templateType: string) {
        return request.get<any, TestReportTemplateResponseDTO>('/v1/test-report-templates/default', {
            params: { templateType }
        })
    }
}

File: backend-java/test-design-assistant-core/src/main/java/com/sinosoft/testdesign/entity/TestReportTemplate.java
package com.sinosoft.testdesign.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 测试报告模板实体
 * 
 * @author sinosoft
 * @date 2024-01-17
 */
@Data
@Entity
@Table(name = "test_report_template")
public class TestReportTemplate {
    
    /**
     * 主键ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 模板编码（TMP-YYYYMMDD-序号）
     */
    @Column(name = "template_code", unique = true, nullable = false, length = 100)
    private String templateCode;
    
    /**
     * 模板名称
     */
    @Column(name = "template_name", nullable = false, length = 500)
    private String templateName;
    
    /**
     * 模板类型：EXECUTION/COVERAGE/QUALITY/RISK
     */
    @Column(name = "template_type", nullable = false, length = 50)
    private String templateType;
    
    /**
     * 模板内容（JSON格式）
     */
    @Column(name = "template_content", nullable = false, columnDefinition = "TEXT")
    private String templateContent;
    
    /**
     * 模板变量定义（JSON格式）
     */
    @Column(name = "template_variables", columnDefinition = "TEXT")
    private String templateVariables;
    
    /**
     * 文件格式：WORD/PDF/EXCEL
     */
    @Column(name = "file_format", length = 50)
    private String fileFormat;
    
    /**
     * 模板描述
     */
    @Column(name = "template_description", columnDefinition = "TEXT")
    private String templateDescription;
    
    /**
     * 是否默认模板：1-是，0-否
     */
    @Column(name = "is_default", length = 1)
    private String isDefault = "0";
    
    /**
     * 是否启用：1-启用，0-禁用
     */
    @Column(name = "is_active", length = 1)
    private String isActive = "1";
    
    /**
     * 版本号
     */
    @Column(name = "version")
    private Integer version = 1;
    
    /**
     * 创建人ID
     */
    @Column(name = "creator_id")
    private Long creatorId;
    
    /**
     * 创建时间
     */
    @Column(name = "create_time")
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    @Column(name = "update_time")
    private LocalDateTime updateTime;
    
    @PrePersist
    public void prePersist() {
        this.createTime = LocalDateTime.now();
        this.updateTime = LocalDateTime.now();
    }
    
    @PreUpdate
    public void preUpdate() {
        this.updateTime = LocalDateTime.now();
    }
}

